sdk/eventhub/event-hubs/samples/typescript/src/iothubConnectionString.ts

```diff
@@ -0,0 +1,166 @@
+    throw new Error(`Invalid IotHub connection string.`);
+  }
+
+  //Extract the IotHub name from the hostname.
+  const [iotHubName] = HostName.split(".");
+
+  if (!iotHubName) {
+    throw new Error(`Unable to extract the IotHub name from the connection string.`);
+  }
+
+  // Generate a token to authenticate to the service.
+  // The code for generateSasToken can be found at https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-tokens
+  const token = generateSasToken(
+    `${HostName}/messages/events`,
+    SharedAccessKey,
+    SharedAccessKeyName,
+    5 // token expires in 5 minutes
+  );
+
+  const connection = new Connection({
+    transport: "tls",
+    host: HostName,
+    hostname: HostName,
+    username: `${SharedAccessKeyName}@sas.root.${iotHubName}`,
+    port: 5671,
+    reconnect: false,
+    password: token
+  });
+  await connection.open();
+
+  // Create the receiver that will trigger a redirect error.
+  const receiver = await connection.createReceiver({
+    source: { address: `amqps://${HostName}/messages/events/$management` }
+  });
+
+  return new Promise((resolve, reject) => {
+    receiver.on(ReceiverEvents.receiverError, (context) => {
+      const error = context.receiver && context.receiver.error;
+      if (isAmqpError(error) && error.condition === "amqp:link:redirect") {
+        const hostname = error.info && error.info.hostname;
+        if (!hostname) {
+          reject(new Error(`Unable to convert connection string.`));
+        } else {
+          resolve(
+            `Endpoint=sb://${hostname}/;EntityPath=${iotHubName};SharedAccessKeyName=${SharedAccessKeyName};SharedAccessKey=${SharedAccessKey}`
+          );
+        }
+      } else {
+        reject(new Error(`Unable to convert connection string.`));
+      }
+      connection.close().catch(() => {
+        /* ignore error */
+      });
+    });
+  });
+}
+
+async function main() {
+  console.log(`Running iothubConnectionString sample`);
+
+  const eventHubsConnectionString = await convertIotHubToEventHubsConnectionString(
+    "HostName=<your-iot-hub>.azure-devices.net;SharedAccessKeyName=<KeyName>;SharedAccessKey=<Key>"
+  );
+
+  const consumerClient = new EventHubConsumerClient(consumerGroup, eventHubsConnectionString);
+
+  const subscription = consumerClient.subscribe(
+    {
+      // The callback where you add your code to process incoming events
+      processEvents: async (events, context) => {
+        for (const event of events) {
+          console.log(
+            `Received event: '${event.body}' from partition: '${context.partitionId}' and consumer group: '${context.consumerGroup}'`
+          );
+        }
+      },
+      processError: async (err, context) => {
+        console.log(`Error : ${err}`);
+      }
+    },
+    { startPosition: earliestEventPosition }
+  );
+
+  // Wait for a bit before cleaning up the sample
+  setTimeout(async () => {
```

Comment:
I feel like I've seen this pattern exported as a function somewhere....(never mind)